tuning/trainercontroller/operations/hfcontrols.py

```diff
@@ -0,0 +1,39 @@
+# Standard
+import re
+import inspect
+from dataclasses import fields
+
+# Third Part
+from transformers import TrainerControl
+
+# Local
+from tuning.trainercontroller.operations import Operation
+
+class HFControls(Operation):
+    """Implements the control actions for the HuggingFace controls in 
+    transformers.TrainerControl class."""
+
+    def __init__(self, **kwargs):
+        """Initializes the HuggingFace controls. In this init, the fields with `should_` of the
+        transformers.TrainerControl data class are extracted, and for each of those fields, the
+        control_action() method's pointer is set, and injected as a class member function.
+
+        Args:
+            kwargs: List of arguments (key, value)-pairs
+        """
+        for control_field in fields(TrainerControl):
+            if re.search(r'^should_', control_field.name) is not None:
```

Comment:
@alex-jw-brooks The reason we went with a regex for the following reasons:
1. It is to avoid name definitions like `should_` (which seems to be a valid function name in python)
2. To leave the door open for newer operations to be added later (in which case, we simply expand the scope of regex accordingly).
3. Also, as a side note, this is a init time operation, and the regex will not bog down the trainer loop.

Reply:
Cool, just wanted to check since `^should_` also allows `should_` - I see it's been changed now, sounds good to me, thanks!